core/client/verify_test.go

```diff
@@ -16,26 +16,51 @@ package client
 
 import (
 	"context"
+	"encoding/json"
+	"io/ioutil"
 	"testing"
 
+	"github.com/google/keytransparency/core/testdata"
+
+	pb "github.com/google/keytransparency/core/api/v1/keytransparency_proto"
 	_ "github.com/google/trillian/merkle/coniks"    // Register hasher
 	_ "github.com/google/trillian/merkle/objhasher" // Register hasher
+	"github.com/google/trillian/types"
 )
 
-// Test vectors were obtained by observing the integration tests, in particular by adding logging
-// output around the calls to GetEntry and VerifyGetEntryResponse in grpc_client.go, and the input
-// to merkle.VerifyMapInclusionProof in VerifyGetEntryResponse.
+// Test vectors in core/testdata are generated by running
+// TRILLIAN_SQL_DRIVER=mysql go test ./impl/integration/ -run TestIntegration/TestEmptyGetAndUpdate
```

Comment:
I don't really get this, but likely I just need it explaining a little more.

You have made a function - SaveTestVectors - that will write test vectors to domain.json and getentryresponse.json.  You have to fun the TestEmptyGetAndUpdate test to get this function to run.  However, in this PR you are also checking in the data that is saved after TestEmptyGetAndUpdate (and SaveTestVectors within that function) is run?

I guess I'm wondering whether you expect the test vectors written to domain.json and getentryresponse.json to be re-generated and overwritten at any point?  Because if not, why not just check in the json files, and if you want to check in the code you used to generate them then separate that out from the tests completely (so that one of the tests doesn't re-write these files each time it's run)?

Will something different be generated/written each time TestEmptyGetAndUpdate is run?  Is the plan that this is run multiple times to get different test vectors each time, and test using the different ones?  I guess a little more explanation as to the purpose of this PR and how it'll work/be used would be useful to help me understand :)

Reply:
Good questions - and the confusion is mine, not yours. 
- This code should probably be factored out to it's own "generate test vectors" program. 
  -  This code would have a lot of overlap with the existing tests, hence the initial desire to stuff this in the tests.
- The generated test vectors are not idempotent.  Things like timestamps and nonce's used in signatures make different runs generate different test vectors.

Reply:
I seeee :)  Thank you for making the changes to make this clearer!  You could always refactor out the bits of code & test cases that are common between this test function and the generation tool and put them somewhere for use in both this test and the generation tool?  I'll leave that decision up to you though :)

Either way, you should probably update/remove this comment now though, as it's no longer accurate :P